dwi_ml/data/hdf5_creation.py

```diff
@@ -0,0 +1,242 @@
+        nib.save(group_image,
+                 str(subj_output_path.joinpath(output_fname)))
+
+    # Standardize data
+    logging.debug('  Standardizing data')
+    standardized_group_data = standardize_data(group_data, subj_mask_data)
+
+    # Save standardized data
+    if save_intermediate:
+        logging.debug('  Saving intermediate standardized files.')
+        standardized_img = nib.Nifti1Image(standardized_group_data,
+                                           group_affine)
+        output_fname = "{}_standardized.nii.gz".format(group)
+        nib.save(standardized_img,
+                 str(subj_output_path.joinpath(output_fname)))
+
+    return standardized_group_data, group_affine, group_header
+
+
+def process_streamlines(bundles_dir: Path, bundles,
+                        header: nib.Nifti1Header, step_size: float,
+                        space: Space):
+    """Load and process a group of bundles and merge all streamlines
+    together.
+
+    Parameters
+    ----------
+    bundles_dir : Path
+        Path to bundles folder.
+    bundles: List[str]
+        List of the bundles filenames to load.
+    header : nib.Nifti1Header
+        Reference used to load and send the streamlines in voxel space and to
+        create final merged SFT.
+    step_size: float
+        Step size to resample streamlines. If none, compress streamlines.
+    space: Space
+        Space to place the tractograms.
+
+    Returns
+    -------
+    output_tractogram : StatefulTractogram
+        All streamlines in voxel space.
+    output_lengths : List[float]
+        The euclidean length of each streamline
+    """
+    # Silencing SFT's logger if our logging is in DEBUG mode
+    set_sft_logger_level('WARNING')
+
+    # Initialize
+    output_tractogram = None
+    output_lengths = []
+
+    # Find official bundles list
+    if bundles is None:
+        # Take everything found in subject bundle folder
+        bundles = [str(p) for p in bundles_dir.glob('*')]
+        if len(bundles) == 0:
+            raise ValueError("No bundle found in the bundles folder!")
+
+    for bundle_name in bundles:
+        # Find bundle name
+        # Note. glob uses str. Other possibility: Path.cwd().glob but creates
+        # a generator object.
+        logging.debug('    Loading bundle {}'.format(bundle_name))
+        bundle_possible_name = str(bundles_dir.joinpath('*' +
+                                                        bundle_name + '*'))
+        bundle_real_name = glob.glob(bundle_possible_name)
+        if len(bundle_real_name) == 0:
+            raise ValueError('Bundle {} not found!'.format(bundle_name))
+        elif len(bundle_real_name) > 1:
+            raise ValueError(
+                'More than one file with name {}. Clean your bundles '
+                'folder.'.format(bundle_possible_name))
+
+        # Loading bundle and sending to wanted space
+        bundle = load_tractogram(bundle_real_name[0], header)
```

Comment:
I would actually add a check before for the extension to see if the filename is a tractogram